### PR TITLE
Fix axial positional encoding calculations for  reformer.mdx

### PR DESCRIPTION
**Pull request by ijindal**
Fix axial positional encoding calculations

# What does this PR do?
This PR corrects the calculations for Axial Positional Encoding in Reformer model documentation. 

- Since d = d_1 + d_2
- if d = 2^10 = 1024
- then 
- d_1 and d_2 both should not be equal to 2^5. As 2^5 + 2^5 = 32+32 = 64 not equal to 1024. 
- d_1 and d_2 should sum to 2^10. Therefore I fixed the d_1 and d_2 dimensions to be equal to 2^9. 
- that is 2^9 + 2^9 = 1024.

and fixed the subsequent calculations. 

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by sgugger** [comment]
cc @ArthurZucker

File: docs/source/en/model_doc/reformer.mdx [hunk 1]
```diff
@@ -83,8 +83,8 @@ factorized embedding vectors: \\(x^1_{k, l} + x^2_{l, k}\\), where as the `confi
 \\(j\\) is factorized into \\(k \text{ and } l\\). This design ensures that each position embedding vector
 \\(x_j\\) is unique.
 
-Using the above example again, axial position encoding with \\(d^1 = 2^5, d^2 = 2^5, n_s^1 = 2^9, n_s^2 = 2^{10}\\)
-can drastically reduced the number of parameters to \\(2^{14} + 2^{15} \approx 49000\\) parameters.
+Using the above example again, axial position encoding with \\(d^1 = 2^9, d^2 = 2^9, n_s^1 = 2^9, n_s^2 = 2^{10}\\)
+can drastically reduced the number of parameters to \\(2^{18} + 2^{19} \approx 780000\\) parameters.
```
Comment (ArthurZucker) [inline docs/source/en/model_doc/reformer.mdx:87]:
```suggestion
Using the above example again, axial position encoding with \\(d^1 = 2^9, d^2 = 2^9, n_s^1 = 2^9, n_s^2 = 2^{10}\\)
can drastically reduced the number of parameters from 500 000 000 to \\(2^{18} + 2^{19} \approx 780 000\\) parameters, this means 85% less memory usage.
```
would just add the previous numbers as comparison for readability.

**Review by ArthurZucker — APPROVED** [review]
LGTM thanks for fixing the doc

File: docs/source/en/model_doc/reformer.mdx (hunk 1 above)
Comment (ijindal) [reply docs/source/en/model_doc/reformer.mdx]: Make sense. Great!

**Pull request merged**